roles/common/tasks/main.yml

```diff
@@ -1,111 +1,90 @@
 ---
-- name: common | Confirm that hostname matches inventory
+- name: Common | Confirm that hostname matches inventory
   ansible.builtin.hostname:
     name: "{{ inventory_hostname }}"
   when:
     "ansible_hostname not in inventory_hostname"
   tags: "update_hostname"
-- name: common | Add apt HTTPS capabilities.
-  ansible.builtin.package:
-    name: "{{ item }}"
-    state: present
-  loop:
-    - ca-certificates
-    - apt-transport-https
-  when:
-    - ansible_os_family == "Debian"
 
-- name: common | install snapd
-  ansible.builtin.package:
-    name: "{{ item }}"
-    state: present
+- name: Common | Install Common packages
+  ansible.builtin.include_tasks: "{{ item }}"
   loop:
-    - snapd
-  when: running_on_server
-
-- name: common | Install procs
-  community.general.snap:
-    name:
-      - procs
-  when: running_on_server
-
-- name: common | install building software and build essentials
-  ansible.builtin.package:
-    name: "{{ item }}"
-    state: present
-    update_cache: true
-  with_items: "{{ common_packages }}"
+    - ubuntu.yml
+    - rockylinux.yml
+  when:
+    - item is defined
+    - item is not none
 
-- name: common | copy tmux.conf
+- name: Common | copy tmux.conf
   ansible.builtin.template:
     src: "tmux.conf.j2"
     dest: "/etc/tmux.conf"
-    mode: 0644
+    mode: "0644"
 
-- name: common | create vim directory
+- name: Common | create vim directory
   ansible.builtin.file:
     path: /etc/vim
     state: directory
+    mode: "0644"
 
-- name: common | copy vimrc
+- name: Common | copy vimrc
   ansible.builtin.copy:
     src: files/vimrc.local
     dest: /etc/vim/vimrc.local
-    mode: 0644
+    mode: "0644"
 
-- name: common | ensure logrotate timer dir exists
+- name: Common | ensure logrotate timer dir exists
   ansible.builtin.file:
     path: /etc/systemd/system/logrotate.timer.d
     state: directory
-    mode: 0755
+    mode: "0755"
     owner: root
     group: root
   when: ansible_service_mgr=="systemd"
 
-- name: common | override default logrotate timer
+- name: Common | override default logrotate timer
   ansible.builtin.template:
     src: "logrotate_timer.j2"
     dest: "/etc/systemd/system/logrotate.timer.d/override.conf"
-    mode: 0644
+    mode: "0644"
   when: ansible_service_mgr=="systemd"
   notify:
     - reload logrotate timer settings
 
-- name: common | install configured dependencies
+- name: Common | install configured dependencies
   ansible.builtin.package:
     name: '{{ configured_dependencies }}'
     state: present
 
-- name: common | install duf
-  ansible.builtin.apt:
-    deb: "{{ duf_url }}/v{{ duf_version }}/duf_{{ duf_version }}_linux_amd64.deb"
-  register: duf_repo
-  retries: 3
-  delay: 60
-  until: duf_repo is succeeded
+- name: Common | install duf
+  ansible.builtin.package:
+    state: present
   when:
-    - running_on_server
-    - ansible_os_family == "Debian"
+    - ansible_distribution == "Ubuntu"
+    - ansible_distribution_release == "jammy"
```

Comment:
Could we add this to the ubuntu package list, instead of having a separate task for it?

Reply:
I should stop caring about bionic.